File: algorithms/[643]子数组最大平均数 I/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
const findMaxAverage = function(nums, k) {
  const len = nums.length;
  if (!len) return 0;

  let sum = nums.slice(0, k).reduce((a, b) => a + b);
  let res = sum;

  for (let i = k; i < len; i += 1) {
    sum += nums[i] - nums[i - k];
    res = Math.max(sum, res);
  }

  return res / k;
};

File: algorithms/[643]子数组最大平均数 I/solution.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)
        res = total = sum(nums[:k])

        for i in range(1, n - k + 1):
            total = total + nums[i + k - 1] - nums[i - 1]
            res = max(res, total)

        return res / k
